ledger/shelley.go

```diff
@@ -397,6 +397,51 @@ func (t *ShelleyTransaction) Cbor() []byte {
 	return cborData
 }
 
+type ShelleyProtocolParameters struct {
+	cbor.StructAsArray
+	MinFeeA            uint
+	MinFeeB            uint
+	MaxBlockBodySize   uint
+	MaxTxSize          uint
+	MaxBlockHeaderSize uint
+	KeyDeposit         uint
+	PoolDeposit        uint
+	MaxEpoch           uint
+	NOpt               uint
+	A0                 *cbor.Rat
+	Rho                *cbor.Rat
+	Tau                *cbor.Rat
+	ProtocolMajor      uint
+	ProtocolMinor      uint
+	Decentralization   *cbor.Rat
+	Nonce              *cbor.Rat
+	ProtocolVersion    []uint
```

Comment:
This would instead be two top level fields for major/minor like we have for Babbage...for reasons